haystack/tools/parameters_schema_utils.py

```diff
@@ -0,0 +1,107 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import collections
+from dataclasses import MISSING, fields, is_dataclass
+from inspect import getdoc
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, Union, get_args, get_origin
+
+from pydantic import Field, create_model
+
+from haystack import logging
+from haystack.dataclasses import ChatMessage
+from haystack.lazy_imports import LazyImport
+from haystack.tools.errors import SchemaGenerationError
+from haystack.tools.from_function import _remove_title_from_schema
+
+with LazyImport(message="Run 'pip install docstring-parser'") as docstring_parser_import:
+    from docstring_parser import parse
+
+
+logger = logging.getLogger(__name__)
+
+
+def _get_param_descriptions(method: Callable) -> Tuple[str, Dict[str, str]]:
+    """
+    Extracts parameter descriptions from the method's docstring using docstring_parser.
+
+    :param method: The method to extract parameter descriptions from.
+    :returns: A dictionary mapping parameter names to their descriptions.
+    """
+    docstring = getdoc(method)
+    if not docstring:
+        return "", {}
+
+    docstring_parser_import.check()
+    parsed_doc = parse(docstring)
+    param_descriptions = {}
+    for param in parsed_doc.params:
+        if not param.description:
+            logger.warning(
+                "Missing description for parameter '%s'. Please add a description in the component's "
+                "run() method docstring using the format ':param %%s: <description>'. "
+                "This description helps the LLM understand how to use this parameter." % param.arg_name
+            )
+        param_descriptions[param.arg_name] = param.description.strip() if param.description else ""
+    return parsed_doc.short_description or "", param_descriptions
+
+
+def _dataclass_to_pydantic_model(dc_type: Any) -> Any:
```

Comment:
nit: can we use a more specific type hinting here?
https://docs.pydantic.dev/latest/api/base_model/#pydantic.create_model